pkg/commands/docker.go

```diff
@@ -205,6 +205,14 @@ func PopulateDigests(runner build.Runner, dependencies []build.ImageDependencies
 func queryDigest(runner build.Runner, reference *build.ImageReference) error {
 	if reference != nil {
 		refString := reference.String()
+
+		// refString will always have the tag specified at this point.
+		// For "scratch", we have to compare it against "scratch:latest" even though
+		// scratch:latest isn't valid in a FROM clause.
+		if refString == build.NormalizeImageTag(constants.NoBaseImageSpecifier) {
```

Comment:
Since it's a special case and you already put very detailed comments, do you think 
`if refString == "scrath:latest"` is more clear?